worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionDataWriter.java

```diff
@@ -416,7 +416,13 @@ public void write(ByteBuf data) throws IOException {
       }
 
       data.retain();
-      flushBuffer.addComponent(true, data);
+      try {
+        flushBuffer.addComponent(true, data);
```

Comment:
Only the non-memory shuffle file storage type requires a call to releaseDiskBuffer?